Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -222,15 +222,56 @@ private func createResolvedPackages(
         return ($0.package.identity, $0)
     }
 
+    // Gather all module aliases declared for dependencies from each package
+    var moduleAliasMap = [String: (String, String)]()
+    packageBuilders.forEach { $0.package.targets.forEach { $0.dependencies.forEach { d in
+        if case let .product(prodRef, _) = d {
+            if let moduleAliasesForDep = prodRef.moduleAliases, let depPkg = prodRef.package {
+                for (key, val) in moduleAliasesForDep {
+                    moduleAliasMap[key] = (val, depPkg)
+                }
+            }
+        }
+    }}}
+  
+    func setModuleAliasesForTargets(in pkg: Package) {
```

Comment:
should this be a private extension of Package instead of a nested function that takes package?